src/module.c

```diff
@@ -3048,6 +3087,227 @@ int RM_HashGet(RedisModuleKey *key, int flags, ...) {
     return REDISMODULE_OK;
 }
 
+/* --------------------------------------------------------------------------
+ * Key API for the stream type.
+ * -------------------------------------------------------------------------- */
+
+/* Adds an entry to a stream. Like XADD without trimming.
+ *
+ * - `key`: The key where the stream is (or will be) stored
+ * - `flags`: A bit field of
+ *   - `REDISMODULE_STREAM_AUTOID`: Assign a stream ID automatically, like `*`
+ *     in the XADD command.
+ *   - `REDISMODULE_STREAM_NOMKSTREAM`: Don't create the stream if it doesn't
```

Comment:
do we really need that `REDISMODULE_STREAM_NOMKSTREAM`?
it's a bit hard to resist, but i think we should try hard to escape from the desire to map the commands "as is" into functions.
and instead come up with a useful API.
in this case i think the module can check the type of the key it opened, and we don't want that flag.

i.e. the flag exists for a command for atomicity.

Reply:
Good point!